cmd/statusd-cli/repl.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+)
+
+// REPL implements the read-eval-print loop for the commands
+// to be sent to statusd.
+type REPL struct {
+	conn   net.Conn
+	reader *bufio.Reader
+	writer *bufio.Writer
+}
+
+// NewREPL creates a REPL instance communicating with the
+// addressed statusd.
+func NewREPL(serverAddress string) (*REPL, error) {
+	conn, err := net.Dial("tcp", serverAddress+":51515")
+	if err != nil {
+		return nil, fmt.Errorf("error when starting REPL: %v", err)
+	}
+	return &REPL{
+		conn:   conn,
+		reader: bufio.NewReader(conn),
+		writer: bufio.NewWriter(conn),
+	}, nil
+}
+
+// Run operates the loop to read a command and its arguments,
+// execute it via the client, and print the result.
+func (r *REPL) Run() error {
+	defer r.conn.Close() // nolint: errcheck
+	input := bufio.NewReader(os.Stdin)
+	for {
```

Comment:
I wonder if there is a library that does that already...

Reply:
Maybe there are, but tried to avoid another external dependency for less than 100 loc. If we get aware of need more convenience or power it surely makes sense to switch.